### PR TITLE
WordAds: Enable A8C CMP banner support

### PR DESCRIPTION
**Pull request by Alameen688**
## Proposed changes:
Adds the support for loading the A8C TCF 2.2 compatible CMP banner on Jetpack sites enrolled in WordAds. This allows collecting consent for personalized ads from site visitors under GDPR laws.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
N/A

## Does this pull request change what data or activity we track or use?
N/A

## Testing instructions:

* Enroll your test Jetpack site to WordAds, if you have not enrolled already.
* Comment out this `wordads_cmp_enabled` check https://github.com/Automattic/jetpack/commit/1987516e4df5f00956ad8bff71a77a63f604df59 (as this option is available in a different branch https://github.com/Automattic/jetpack/pull/35205)
* Visit your test Jetpack site from an EU country (you might need a VPN for this)
* The CMP banner should appear and you should be able to manage your privacy preference.
* Confirm that ad requests respect privacy preference.
cmp-on-jetpack
cmp-on-jetpack-2

**Confirm no conflict with Cookie consent block**
- First, enable the [Extra Sidebar Widgets](https://jetpack.com/support/extra-sidebar-widgets/), so the block can be available on your site.
- Add the cookie consent block to your site using [this guide](https://wordpress.com/support/wordpress-editor/blocks/cookie-consent-block/).
- Visit your site and confirm the Cookie consent block is NOT visible and is overridden by the CMP GDPR banner
- Now deactivate the GDPR banner from running e.g by NOT commenting out this `wordads_cmp_enabled` check this time https://github.com/Automattic/jetpack/commit/1987516e4df5f00956ad8bff71a77a63f604df59 (as this option is available in a different branch https://github.com/Automattic/jetpack/pull/35205)
- Visit your site again and this time the Cookie consent block should be visible
cookie consent

### CODE REVIEW

File: projects/plugins/jetpack/modules/wordads/php/class-wordads-consent-management-provider.php [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+	 * IAB specified cookie name for storing the consent string.
+	 */
+	const COOKIE_NAME = 'euconsent-v2';
+
+	/**
+	 * Initializes loading of the frontend framework.
+	 */
+	public static function init() {
+		// Prevent Cookies & Consent banner from displaying when the CMP is active.
+		add_filter( 'jetpack_disable_eu_cookie_law_widget', '__return_true' );
```
Comment (jeherve) [inline projects/plugins/jetpack/modules/wordads/php/class-wordads-consent-management-provider.php:28]: Should we support our Cookie Consent block too?

Comment (cphilleo) [reply projects/plugins/jetpack/modules/wordads/php/class-wordads-consent-management-provider.php:28]: We'd need to override the Cookie Consent block and prevent it from displaying if the CMP is enabled. Is there a filter in place already for this? If not, I can create one.

Comment (jeherve) [reply projects/plugins/jetpack/modules/wordads/php/class-wordads-consent-management-provider.php:28]: Would you want to block the display of the block on the frontend only, or in the editor as well?

Comment (cphilleo) [reply projects/plugins/jetpack/modules/wordads/php/class-wordads-consent-management-provider.php:28]: The frontend at minimum, but preferably in the editor as well. That would be the least confusing option for users, I think.

Comment (jeherve) [reply projects/plugins/jetpack/modules/wordads/php/class-wordads-consent-management-provider.php:28]: To stop things on the frontend, you could add an early return here if a specific condition is met:
https://github.com/Automattic/jetpack/blob/8447ef5075727869218037724844a6b6ba29b35c/projects/plugins/jetpack/extensions/blocks/cookie-consent/cookie-consent.php#L71

In the editor, I'm guessing you could display a notice somewhere here:
https://github.com/Automattic/jetpack/blob/79a546280f22e957c9792832f414645d50a3eba1/projects/plugins/jetpack/extensions/blocks/cookie-consent/edit.js#L82

**Review by cphilleo — APPROVED** [review]
Looks good!

**Comment by jeherve** [comment]
I gave this a try, and opened #35791 to address some issues I found when testing.

**Pull request merged**